internal/index/indexoperations/index.go

```diff
@@ -52,3 +54,18 @@ func ListIndexes(path string) ([]Index, error) {
 	}
 	return indexes, nil
 }
+
+// AddIndex initializes a new index to install plugins from.
+func AddIndex(path, name, url string) error {
+	if strings.ContainsAny(name, "./") {
```

Comment:
as you might imagine this won't work on Windows

please use os.PathSeparator

Reply:
consider extracting this to `IsValidIndexName` method and check for further things, like spaces, `\` `/`, `.` ... + add unit tests + reuse everywhere (add, delete, ...)

Reply:
Even easier, we could simply restrict index names to k8s resource names (see https://kubernetes.io/docs/concepts/overview/working-with-objects/names/#dns-subdomain-names). Hopefully, there is a k8s tool that we can just use?

Reply:
Like `IsValidPathSegmentName` https://github.com/kubernetes/apimachinery/blob/master/pkg/api/validation/path/name.go?

Reply:
Let's start restrictive, we can relax as we need.
Honestly, something like `[A-Za-z0-9_-]` is good enough and safest for now. Let's make it a method regardless as I'm intending to reuse in other places.

Reply:
Ah, finally found it: https://github.com/kubernetes/kubernetes/blob/fdb2cb4c8832da1499069bda918c014762d8ac05/staging/src/k8s.io/apimachinery/pkg/util/validation/validation.go#L203

If we can, let's use it. Then we get all the goodness with names starting with alpha characters.

Reply:
Are you 100% sure that string supports all repo names on github. If so, let's use it. If not, let's stick to something more relevant.

Reply:
Looks like GitHub also allows dots in repo names, i.e. `[A-Za-z0-9_-.]+`. And it does not care whether a repo name starts with an alpha character or `_-.`. So if we want to support everything that GitHub does, the current pattern is better.